Lambdas/GHOAuth/OAuthLambda.swift

```diff
@@ -0,0 +1,194 @@
+        guard let publicKeyString = ProcessInfo.processInfo.environment["ACCOUNT_LINKING_OAUTH_FLOW_PUB_KEY"] else {
+            throw Errors.envVarNotFound(name: "ACCOUNT_LINKING_OAUTH_FLOW_PUB_KEY")
+        }
+        guard let publicKeyData = Data(base64Encoded: publicKeyString) else {
+            throw Errors.invalidPublicKey
+        }
+        let publicKey = try ECDSAKey.public(pem: publicKeyData)
+
+        let signer = JWTSigner.es256(key: publicKey)
+
+        let payload = try signer.verify(state, as: GHOAuthPayload.self)
+
+        return payload
+    }
+
+    func getGHAccessToken(code: String) async throws -> String {
+        logger.debug("Retrieving GitHub client secrets")
+
+        let clientSecret = try await secretsRetriever.getSecret(arnEnvVarKey: "GH_CLIENT_SECRET_ARN")
+        guard let clientID = ProcessInfo.processInfo.environment["GH_CLIENT_ID"] else {
+            throw Errors.envVarNotFound(name: "GH_CLIENT_ID")
+        }
+
+        // https://docs.github.com/en/apps/oauth-apps/building-oauth-apps/authorizing-oauth-apps
+
+        logger.debug("Requesting GitHub access token")
+        var request = HTTPClientRequest(url: "https://github.com/login/oauth/access_token")
+        request.method = .POST
+        request.headers = [
+            "Accept": "application/json",
+            "Content-Type": "application/json"
+        ]
+        let requestBody = try jsonEncoder.encode([
+            "client_id": clientID,
+            "client_secret": clientSecret,
+            "code": code
+        ])
+        request.body = .bytes(requestBody)
+
+        let response = try await client.execute(request, timeout: .seconds(30))
+        let responseBody = try await response.body.collect(upTo: 1 << 22)
+        logger.debug("Got response \(response.status): headers: \(response.headers), body: \(responseBody)")
+
+        guard response.status == .ok else {
+            throw Errors.badResponse(status: Int(response.status.code))
+        }
+
+        let accessToken = try jsonDecoder.decode(AccessTokenResponse.self, from: responseBody).accessToken
+
+        return accessToken
+    }
+
+    func getGHUser(accessToken: String) async throws -> User {
+        logger.debug("Requesting GitHub user info")
+
+        // https://docs.github.com/en/rest/users/users?apiVersion=2022-11-28#get-the-authenticated-user
+
+        var request = HTTPClientRequest(url: "https://api.github.com/user")
+        request.headers = [
+            "Accept": "application/vnd.github+json",
+            "Authorization": "Bearer \(accessToken)",
+            "X-GitHub-Api-Version": "2022-11-28"
+        ]
+
+        let response = try await client.execute(request, timeout: .seconds(30))
+
+        guard response.status == .ok else {
+            throw Errors.badResponse(status: Int(response.status.code))
+        }
+
+        let userResponseBody = try await response.body.collect(upTo: 1024 * 1024)
```

Comment:
Where are these magic `upTo` numbers coming from in the various calls to `collect()`? Here we have 1MiB, a bit further up we have 4MiB, and in one of the other lambdas I've seen values like 64KiB used.

Reply:
Mostly just what I thought might be good when I wrote the code. Though not really supercalculated or something.